src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -4,1096 +4,686 @@
     {
-        // We only want the parent folder name as the source (not the full path)
-        const auto source = fragmentExtFolder.path().filename().wstring();
-
-        // Only apply the stubs if the parent folder name is not in ignored namespaces
-        // (also make sure this is a directory for sanity)
-        if (std::filesystem::is_directory(fragmentExtFolder) && ignoredNamespaces.find(source) == ignoredNamespaces.end())
+        if (const auto val = json.find(key.data(), key.data() + key.size()))
         {
-            const auto jsonFiles = _AccumulateJsonFilesInDirectory(fragmentExtFolder.path().c_str());
-            _ParseAndLayerFragmentFiles(jsonFiles, winrt::hstring{ source });
+            return *val;
         }
     }
+
+    return Json::Value::nullSingleton();
 }
 
-// Method Description:
-// - Finds all the json files within the given directory
-// Arguments:
-// - directory: the directory to search
-// Return Value:
-// - A set containing all the found file data
-std::unordered_set<std::string> CascadiaSettings::_AccumulateJsonFilesInDirectory(const std::wstring_view directory)
+// Returns true if the given Json::Value looks like a profile.
+// We introduced a bug (GH#9962, fixed in GH#9964) that would result in one or
+// more nameless, guid-less profiles being emitted into the user's settings file.
+// Those profiles would show up in the list as "Default" later.
+bool SettingsLoader::_isValidProfileObject(const Json::Value& profileJson)
 {
-    std::unordered_set<std::string> jsonFiles;
-
-    for (const auto& fragmentExt : std::filesystem::directory_iterator(directory))
-    {
-        if (fragmentExt.path().extension() == jsonExtension)
-        {
-            try
-            {
-                jsonFiles.emplace(ReadUTF8File(fragmentExt.path()));
-            }
-            CATCH_LOG();
-        }
-    }
-    return jsonFiles;
+    return profileJson.isObject() &&
+           (profileJson.isMember(NameKey.data(), NameKey.data() + NameKey.size()) || // has a name (can generate a guid)
+            profileJson.isMember(GuidKey.data(), GuidKey.data() + GuidKey.size())); // or has a guid
 }
 
-// Method Description:
-// - Given a set of json files, uses them to modify existing profiles,
-//   create new profiles, and create new color schemes
-// Arguments:
-// - files: the set of json files (each item in the set is the file data)
-// - source: the location the files came from
-void CascadiaSettings::_ParseAndLayerFragmentFiles(const std::unordered_set<std::string> files, const winrt::hstring source)
+// Parses the given JSON string ("content") and fills a ParsedSettings instance with it.
+void SettingsLoader::_parse(const OriginTag origin, const winrt::hstring& source, const std::string_view& content, ParsedSettings& settings)
 {
-    for (const auto& file : files)
+    static constexpr std::string_view emptyObjectJSON{ "{}" };
+
+    const auto json = _parseJSON(content.empty() ? emptyObjectJSON : content);

```

Comment:
This seems more expensive than...

```suggestion
    const auto json = content.empty() ? Json::Value{ Json::Value::objectValue } : _parseJSON(content);
```

(parsing `{}` to get a Json::Value?)